Automattic/jetpack
# Editor Metadata: Add the last editor used to the post meta

In some circumstances it is beneficial to know which editor has been
used to edit a post. The classic editor plugin already stores this
information so that it can be displayed in the post list, and be used to
initialise the correct editor by default.

#### Changes proposed in this Pull Request:

This adds the same functionality to Jetpack and adds the metadata to
the post if the classic editor plugin will not, either because it's not
active, or because of its settings.

It adds a library of the functions needed, and a tweak to the `wpcom-block-editor` module to hook in the functions as necessary.


#### Does this pull request change what data or activity we track or use?

Not directly, but the data could be used to track activity

#### Testing instructions:
* Create or edit a post and check that the `classic-editor-remember` metadata value is set
* Activate the classic editor plugin and create or update a post. Check again that the metadata value is set
* Change the classic editor plugin settings, and repeat using the editor, checking for the metadata value
* As much as possible check that for any post you create, the correct metadata value is set.

#### Proposed changelog entry for your changes:
I'm not if this change needs a changelog entry. I'll follow advice here, but maybe:

* Adds functionality to attach metadata to a post indicating the last editor used.
